Celeste.Mod.mm/Patches/Decal.cs

```diff
@@ -82,23 +86,51 @@ private class DecalImage : Component {
         [MakeMethodPublic]
         public extern void MakeBanner(float speed, float amplitude, int sliceSize, float sliceSinIncrement, bool easeDown, float offset = 0f, bool onlyIfWindy = false);
 
-        [MonoModIgnore]
+        [MonoModReplace]
         [MakeMethodPublic]
-        public extern void MakeSolid(float x, float y, float w, float h, int surfaceSoundIndex, bool blockWaterfalls = true);
+        public void MakeSolid(float x, float y, float w, float h, int surfaceSoundIndex, bool blockWaterfalls = true) {
+            solid = new Solid(Position + new Vector2(x, y), w, h, true);
+            solid.BlockWaterfalls = blockWaterfalls;
+            solid.SurfaceSoundIndex = surfaceSoundIndex;
+            Scene.Add(solid);
+        }
 
         public void MakeCoreSwap(string coldPath, string hotPath) {
             Add(image = new CoreSwapImage(GFX.Game[coldPath], GFX.Game[hotPath]));
         }
 
         public void MakeStaticMover(int x, int y, int w, int h, bool jumpThrus = false) {
-            StaticMover sm = new StaticMover {
+            staticMover = new StaticMover {
                 SolidChecker = s => s.CollideRect(new Rectangle((int) X + x, (int) Y + y, w, h)),
-                OnMove = v => { X += v.X; Y += v.Y; },
-                OnShake = v => { X += v.X; Y += v.Y; },
+                OnDestroy = () => {
+                    RemoveSelf();
+                    if (solid != null) 
+                        solid.RemoveSelf(); 
```

Comment:
nitpick, but this could use the null-chaining operator (`solid?.RemoveSelf();`) (if only that could also be used for assignment, `solid?.Collidable = Active = Visible = Collidable = false;`)